gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java

```diff
@@ -335,22 +343,25 @@ public void orchestrate(Spec spec, Properties jobProps, long triggerTimestampMil
     Instrumented.updateTimer(this.flowOrchestrationTimer, System.nanoTime() - startTime, TimeUnit.NANOSECONDS);
   }
 
-  public void submitFlowToDagManager(FlowSpec flowSpec, Optional<String> optionalFlowExecutionId) throws IOException, InterruptedException {
+  public void submitFlowToDagManager(FlowSpec flowSpec, DagActionStore.DagAction flowAction) throws IOException, InterruptedException {
     Optional<Dag<JobExecutionPlan>> optionalJobExecutionPlanDag =
-        this.flowCompilationValidationHelper.createExecutionPlanIfValid(flowSpec, optionalFlowExecutionId);
+        this.flowCompilationValidationHelper.createExecutionPlanIfValid(flowSpec,
+            Optional.of(flowAction.getFlowExecutionId()));
     if (optionalJobExecutionPlanDag.isPresent()) {
-      submitFlowToDagManager(flowSpec, optionalJobExecutionPlanDag.get());
+      submitFlowToDagManager(flowSpec, optionalJobExecutionPlanDag.get(), flowAction);
     } else {
       _log.warn("Flow: {} submitted to dagManager failed to compile and produce a job execution plan dag", flowSpec);
       Instrumented.markMeter(this.flowOrchestrationFailedMeter);
     }
   }
 
-  public void submitFlowToDagManager(FlowSpec flowSpec, Dag<JobExecutionPlan> jobExecutionPlanDag)
+  public void submitFlowToDagManager(FlowSpec flowSpec, Dag<JobExecutionPlan> jobExecutionPlanDag,
+      DagActionStore.DagAction launchAction)
       throws IOException {
     try {
-      //Send the dag to the DagManager.
+      // Send the dag to the DagManager and delete the action after persisting it to avoid redundant execution on start up
       this.dagManager.get().addDag(jobExecutionPlanDag, true, true);
+      this.dagActionStore.get().deleteDagAction(launchAction);
```

Comment:
what if the DM hasn't successfully handled it (e.g. the service fails after line 364, but before the DM actions complete)?  would the action/event be lost?

Reply:
overall I recognize we're in an interim phase of our multi-leader journey, where we have yet to integrate the DagMgr w/ the MultiActiveLeaseArbiter... this is only temporary so let's not over-engineer

I'm concerned however that this PR's approach presumes successful handling for non-durable/in-memory state.  a safer pattern would be to remove the LAUNCH action from the DagAction store only after the DM has *finished* launching the flow execution.

Reply:
If `DagManager.addDag` completes then we have persisted ([code](https://github.com/apache/gobblin/blob/35304e9676f82feeb6cc93332707a2bd9cae6b74/gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java#L317)) the `dag` in MySQL `DagStateStore` already and its status and completion will be tracked by existing `DagManager` functionality via series of checkpoints once its [submitted to executor](https://github.com/apache/gobblin/blob/35304e9676f82feeb6cc93332707a2bd9cae6b74/gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java#L1107) or fails or next dag is called etc... We load all dags from state store upon startup and while `DagManager` is active it should keep track of it. In pre-multi-active `DagMgr` I don't believe we should proceed further necessarily. What do you think?

Reply:
that's great news.  we're all set then!  sorry, FSR I thought it merely added the DAG to a queue, and hadn't recalled that it synchronously persists the DAG.